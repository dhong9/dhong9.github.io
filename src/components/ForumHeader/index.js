import React from "react";

// @mui material components
import Toolbar from "@mui/material/Toolbar";
import Link from "@mui/material/Link";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

export default function ForumHeader(props) {
  const { categories } = props;

  return (
    <Toolbar
      component="nav"
      variant="dense"
      sx={{ justifyContent: "space-between", overflowX: "auto" }}
    >
      {categories.map(({ id, name }) => (
        <Link
          color="inherit"
          noWrap
          key={id}
          variant="body2"
          href={"/doughnutRider/categories/" + id}
          sx={{ p: 1, flexShrink: 0 }}
        >
          {name}
        </Link>
      ))}
    </Toolbar>
  );
}

// Typechecking props of ForumHeader
ForumHeader.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      description: PropTypes.string,
    })
  ),
};

ForumHeader.defaultProps = {
  categories: [],
  title: "",
};
