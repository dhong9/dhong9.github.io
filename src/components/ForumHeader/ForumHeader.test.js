import renderer from "react-test-renderer";
import ForumHeader from "components/ForumHeader";

describe("ForumHeader", () => {
  it("renders", () => {
    const categories = [
      { id: 1, name: "Computers", description: "All about computers" },
      { id: 2, name: "Wildlife", description: "All about wildlife" },
    ];
    const component = renderer.create(<ForumHeader categories={categories} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
