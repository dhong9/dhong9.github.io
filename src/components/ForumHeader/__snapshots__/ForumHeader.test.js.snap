// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumHeader renders 1`] = `
<nav
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-14105se-MuiToolbar-root"
>
  <a
    className="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-1a72jx6-MuiTypography-root-MuiLink-root"
    href="/doughnutRider/categories/1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Computers
  </a>
  <a
    className="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-1a72jx6-MuiTypography-root-MuiLink-root"
    href="/doughnutRider/categories/2"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Wildlife
  </a>
</nav>
`;
